File: samples/Test/Classes/Control/CheckboxTest.cpp
//
//  CheckboxTest.cpp
//  Test
//
//  Created by ly≤     on 2017/3/1.
//
//

#include "CheckboxTest.h"

CheckboxTest::CheckboxTest()
{
    CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
    drawer->setTouchMoved(false);
}

CheckboxTest::~CheckboxTest()
{
    CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
    drawer->setTouchMoved(true);
}

void CheckboxTest::viewDidLoad()
{
    this->getView()->setColor(CAColor_gray);
    
    CACheckbox* box = CACheckbox::createWithLayout(DLayout(DHorizontalLayout_W_C(54, 0.5), DVerticalLayout_H_C(54, 0.25f)));
    this->getView()->addSubview(box);
    box->setTarget([](bool var)
    {
        CCLog("box: %d", var);
    });
    
    CACheckbox* box1 = CACheckbox::createWithLayout(DLayout(DHorizontalLayout_W_C(240, 0.5), DVerticalLayout_H_C(54, 0.55f)));
    box1->setImageStateNormal(CAImage::create("source_material/switch_on.png"));
    box1->setImageStateSelected(CAImage::create("source_material/switch_off.png"));
    box1->setTitleStateNormal("未选中");
    box1->setTitleStateSelected("选中");
    this->getView()->addSubview(box1);
    box1->setTarget([](bool var)
    {
        CCLog("box: %d", var);
    });
}

void CheckboxTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/CheckboxTest.js
var CheckboxTest = ca.CAViewController.extend({
    ctor: function () {
        this._super();
        this.getView().setColor(ca.color._getGray());

        var box = ca.CACheckbox.createWithLayout(DLayout(DHorizontalLayout_W_C(54, 0.5), DVerticalLayout_H_C(54, 0.25)));
        this.getView().addSubview(box);
        box.setTarget(function ( isSelect) {
          if (isSelect) {
            log("选中");
          }else {
            log("取消选中");
          }
        });
        var box1 = ca.CACheckbox.createWithLayout(DLayout(DHorizontalLayout_W_C(240, 0.5), DVerticalLayout_H_C(54, 0.55)));
        box1.setImageStateNormal(ca.CAImage.create("source_material/checkbox_normal.png"));
        box1.setImageStateSelected(ca.CAImage.create("source_material/checkbox_selected.png"));
        box1.setTitleStateNormal("未选中");
        box1.setTitleStateSelected("选中");
        this.getView().addSubview(box1);
        box1.setTarget(function ( isSelect) {
          if (isSelect) {
            log("选中");
          }else{
            log("取消选中");
          }
        });
    },
});
